src/k8s/pkg/component/loadbalancer.go

```diff
@@ -0,0 +1,129 @@
+package component
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/canonical/k8s/pkg/snap"
+	"github.com/canonical/k8s/pkg/utils/k8s"
+)
+
+func EnableLoadBalancerComponent(s snap.Snap, cidrs []string, l2Enabled bool, l2Interfaces []string, bgpEnabled bool, bgpLocalASN int, bgpPeerAddress string, bgpPeerASN int, bgpPeerPort int) error {
+	manager, err := NewHelmClient(s, nil)
+	if err != nil {
+		return fmt.Errorf("failed to get component manager: %w", err)
+	}
+
+	networkValues := map[string]any{
+		"l2announcements": map[string]any{
+			"enabled": l2Enabled,
+		},
+		"bgpControlPlane": map[string]any{
+			"enabled": bgpEnabled,
+		},
+		"externalIPs": map[string]any{
+			"enabled": true,
+		},
+		// https://docs.cilium.io/en/v1.14/network/l2-announcements/#sizing-client-rate-limit
+		// Assuming for 50 LB services
+		"k8sClientRateLimit": map[string]any{
+			"qps":   10,
+			"burst": 20,
+		},
+	}
+
+	err = manager.Refresh("network", networkValues)
+	if err != nil {
```

Comment:
also in other places in this file, when you don't need other returned values

```suggestion
	if err := manager.Refresh("network", networkValues); err != nil {
```